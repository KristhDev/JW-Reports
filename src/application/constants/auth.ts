import { ItemOption, Precursor } from '@infrasturcture/interfaces';

/**
 * The precursors options to select
 */
export const PRECURSORS_OPTIONS: ItemOption[] = [
    { label: 'Ninguno', value: 'ninguno' },
    { label: 'Auxiliar', value: 'auxiliar' },
    { label: 'Regular', value: 'regular' },
    { label: 'Especial', value: 'especial' }
];

export const precursors: Record<Uppercase<Precursor>, Precursor> = {
    AUXILIAR: 'auxiliar',
    ESPECIAL: 'especial',
    NINGUNO: 'ninguno',
    REGULAR: 'regular'
}

export const authMessages = {
    CONFIRM_PASSWORD_EMPTY: 'La confirmación de la contraseña es requerida.',
    EMAIL_ALREADY_REGISTERED: 'Ya existe un usuario con este correo.',
    EMAIL_EMPTY: 'El correo no puede estar vacío.',
    EMAIL_INVALID: 'Correo electrónico inválido.',
    EMAIL_UPDATE_UNCHANGED: 'Para actualizar tu correo debes cambiarlo.',
    NAME_EMPTY: 'El nombre no puede estar vacío.',
    NAME_MIN_LENGTH: 'El nombre debe tener al menos 2 caracteres.',
    PASSWORD_EMPTY: 'La contraseña no puede estar vacía.',
    PASSWORD_MIN_LENGTH: 'La contraseña debe tener al menos 6 caracteres.',
    PASSWORD_NOT_MATCH: 'Las contraseñas no coinciden.',
    PASSWORD_UPDATED: 'Ha actualizado su contraseña correctamente.',
    PROFILE_UPDATED: 'Ha actualizado su perfil correctamente.',
    SURNAME_EMPTY: 'Los apellidos no pueden estar vacíos.',
    SURNAME_MIN_LENGTH: 'Los apellidos deben tener al menos 2 caracteres.',
    UNATHENTICATED: 'Para realizar está acción debe iniciar sesión.',
    UNAUTHORIZED: 'No tiene permiso para realizar está acción.',
}

export const precursorMessages = {
    PRECURSOR_EMPTY: 'El campo precursor es requerido.',
    PRECURSOR_INVALID: 'Por favor seleccione una opción de precursor.',
}