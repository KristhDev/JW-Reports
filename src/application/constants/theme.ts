import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];