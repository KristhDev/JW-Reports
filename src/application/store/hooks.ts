import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();