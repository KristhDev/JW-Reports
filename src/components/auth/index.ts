export { CredentialsForm } from './CredentialsForm';
export { ForgotPasswordForm } from './ForgotPasswordForm';
export { LoginForm } from './LoginForm';
export { ProfileForm } from './ProfileForm';
export { RegisterForm } from './RegisterForm';