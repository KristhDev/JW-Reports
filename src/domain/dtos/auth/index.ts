export * from './sign-up.dto';
export * from './update.dto';