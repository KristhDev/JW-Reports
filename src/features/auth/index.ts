export { default as authReducer, setUser, clearAuth, setIsAuthLoading } from './slice';