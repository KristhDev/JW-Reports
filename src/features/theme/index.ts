export { default as themeReducer, setTheme } from './slice';