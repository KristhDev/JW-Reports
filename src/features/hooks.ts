import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;