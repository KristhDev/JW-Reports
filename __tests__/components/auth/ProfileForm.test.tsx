import React from 'react';
import { act, fireEvent, render, screen, waitFor } from '@testing-library/react-native';

/* Components */
import { ProfileForm } from '../../../src/components/auth';

/* Hooks */
import { useAuth, useStatus, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Mocks */
import { testUser, updateProfileMock } from '../../mocks';

const setErrorFormMock = jest.fn();

/* Hooks */
jest.mock('../../../src/hooks/useAuth.ts');
jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <ProfileForm /> component', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    (useAuth as jest.Mock).mockReturnValue({
        state: { user: testUser, isAuthLoading: false },
        updateProfile: updateProfileMock
    });

    (useStatus as jest.Mock).mockReturnValue({
        setErrorForm: setErrorFormMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors }
    });

    it('should to match snapshot', async () => {
        await waitFor(() => {
            render(<ProfileForm />);
        });

        await act(() => {
            expect(screen.toJSON()).toMatchSnapshot();
        });
    });

    it('should call setErrorForm when form is invalid', async () => {
        await waitFor(() => {
            render(<ProfileForm />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get text inputs to type empty content */
                const inputsText = screen.getAllByTestId('form-field-text-input');
                fireEvent(inputsText[0], 'onChangeText', '');

                /* Get touchable to submit form */
                const touchable = screen.getByTestId('button-touchable');
                fireEvent.press(touchable);

                /* Check if setErrorForm is called one time */
                expect(setErrorFormMock).toHaveBeenCalledTimes(1);
            });
        });
    });

    it('should call updateProfile when the form is valid', async () => {
        await waitFor(() => {
            render(<ProfileForm />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get touchable to submit form */
                const touchable = screen.getByTestId('button-touchable');
                fireEvent.press(touchable);

                /* Check if updateProfile is called one time */
                expect(updateProfileMock).toHaveBeenCalledTimes(1);
            });

            /* Check if updateProfile is called with respective args */
            expect(updateProfileMock).toHaveBeenCalledWith({
                name: testUser.name,
                surname: testUser.surname,
                precursor: testUser.precursor,
                hoursRequirement: testUser.hoursRequirement
            }, expect.any(Object));
        });
    });
});