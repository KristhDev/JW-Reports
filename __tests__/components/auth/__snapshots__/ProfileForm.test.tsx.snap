// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ProfileForm /> component should to match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "display": "flex",
        "marginBottom": 16,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
      testID="form-field-label"
    >
      Nombre:
    </Text>
    <View
      style={
        {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 10,
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            cursorColor="#C0A7E1"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSelectionChange={[Function]}
            placeholder="Ingrese su nombre"
            placeholderTextColor="#BFBFBF"
            selection={
              {
                "end": 9,
                "start": 9,
              }
            }
            selectionColor="#A0B9E2"
            style={
              [
                {
                  "color": "#FFFFFF",
                  "flex": 1,
                  "fontSize": 17,
                  "paddingLeft": 10,
                  "paddingRight": 5,
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
            testID="form-field-text-input"
            value="Celestino"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#BFBFBF",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "display": "flex",
        "marginBottom": 16,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
      testID="form-field-label"
    >
      Apellidos:
    </Text>
    <View
      style={
        {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 10,
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            cursorColor="#C0A7E1"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSelectionChange={[Function]}
            placeholder="Ingrese su apellido"
            placeholderTextColor="#BFBFBF"
            selection={
              {
                "end": 9,
                "start": 9,
              }
            }
            selectionColor="#A0B9E2"
            style={
              [
                {
                  "color": "#FFFFFF",
                  "flex": 1,
                  "fontSize": 17,
                  "paddingLeft": 10,
                  "paddingRight": 5,
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
            testID="form-field-text-input"
            value="Wilderman"
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#BFBFBF",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "display": "flex",
        "marginBottom": 16,
        "width": 675,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
      testID="form-select-label"
    >
      Precursor:
    </Text>
    <View
      style={
        {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 0,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderRadius": 5,
                "flex": 1,
              }
            }
            testID="form-select-touchable"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "padding": 10,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "flex": 1,
                      "fontSize": 17,
                    },
                    undefined,
                  ]
                }
                testID="form-select-text-value"
              >
                Ninguno
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#BFBFBF",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    statusBarTranslucent={true}
    transparent={true}
    visible={false}
  />
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        false,
        [
          {
            "backgroundColor": "#C0A7E1",
            "borderRadius": 10,
          },
          {
            "marginVertical": 30,
          },
        ],
      ]
    }
    testID="button-touchable"
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "minWidth": "100%",
            "paddingHorizontal": 20,
            "paddingVertical": 9,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "color": "#121212",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 2,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
        testID="button-text"
      >
        Guardar
      </Text>
    </View>
  </View>
</View>
`;
